### PR TITLE
Fixes for various PKCS7 and SRP build issues

### PR DESCRIPTION
**Pull request by dgarske**
* Fix for static memory with bucket size matching.
* Fixes for SRP with heap hint.
* Support for PKCS7 without AES CBC.
* Added GCC-ARM TLS server example.
* Fixes for building without AES CBC.
ZD 12586

### CODE REVIEW

**Comment by dgarske** [comment]
retest this please

File: wolfcrypt/test/test.c [hunk 1]
```diff
@@ -17206,56 +17207,57 @@ WOLFSSL_TEST_SUBROUTINE int srp_test(void)
     byte serverProof[SRP_MAX_DIGEST_SIZE]; /* M2 */
 #endif
 
+    /* determine SRP hash to test */
+#if !defined(NO_SHA256)
+    srpHash = SRP_TYPE_SHA256;
+#elif !defined(N_SHA)
```
Comment (JacobBarthelmeh) [inline wolfcrypt/test/test.c:17213]: typo for NO_SHA

**Review by JacobBarthelmeh — COMMENTED** [review]
The changes fix build of the main library with --enable-pkcs7 --disable-aescbc but there is some issues around the test cases compiling. 

```
tests/api.c:25517:19: error: implicit declaration of function 'wc_AesCbcDecrypt' 
```

and 

```
wolfcrypt/test/test.c:30650:10: warning: unused variable 'secretKey' [-Wunused-variable]
    byte secretKey[] = {
         ^
wolfcrypt/test/test.c:30656:10: warning: unused variable 'secretKeyId' [-Wunused-variable]
    byte secretKeyId[] = {
         ^
wolfcrypt/test/test.c:30643:10: warning: unused variable 'optionalUkm' [-Wunused-variable]
    byte optionalUkm[] = {
         ^
wolfcrypt/test/test.c:30656:10: warning: unused variable 'secretKeyId' [-Wunused-variable]
    byte secretKeyId[] = {
         ^
wolfcrypt/test/test.c:30650:10: warning: unused variable 'secretKey' [-Wunused-variable]
    byte secretKey[] = {
         ^
wolfcrypt/test/test.c:30643:10: warning: unused variable 'optionalUkm' [-Wunused-variable]
    byte optionalUkm[] = {
         ^

```

could you resolve those please in order to do some run time tests with the build.

**Comment by JacobBarthelmeh** [comment]
ok to test

**Comment by dgarske** [comment]
retest this please

File: wolfcrypt/src/srp.c [hunk 2]
```diff
@@ -217,7 +217,7 @@ static void SrpHashFree(SrpHash* hash)
 }
 
 
-int wc_SrpInit(Srp* srp, SrpType type, SrpSide side)
+int wc_SrpInit_ex(Srp* srp, SrpType type, SrpSide side, void* heap, int devId)
```
Comment (SparkiDev) [inline wolfcrypt/src/srp.c:220]: SrpHashInit() is called and it creates a hash without a heap parameter.

**Review by SparkiDev — CHANGES_REQUESTED** [review]
./configure --enable-all --disable-aescbc came up with more issues.

File: src/ssl.c [hunk 3]
```diff
@@ -33894,8 +33894,13 @@ int wolfSSL_CMAC_Init(WOLFSSL_CMAC_CTX* ctx, const void *key, size_t keyLen,
 
     WOLFSSL_ENTER("wolfSSL_CMAC_Init");
 
-    if (ctx == NULL || cipher == NULL || (cipher != EVP_AES_128_CBC &&
-        cipher != EVP_AES_192_CBC && cipher != EVP_AES_256_CBC)) {
+    if (ctx == NULL || cipher == NULL 
+    #ifdef HAVE_AES_CBC
+        || (cipher != EVP_AES_128_CBC &&
```
Comment (SparkiDev) [inline src/ssl.c:33899]: It looks like CMAC requires AES-CBC.
This should fail if AES-CBC is not compiled in.

**Review by SparkiDev — APPROVED** [review]

**Review by SparkiDev — CHANGES_REQUESTED** [review]
Forgot there was a change requested.

File: src/ssl.c (hunk 3 above)
Comment (dgarske) [reply src/ssl.c:33899]: Looks like CMAC requires AES direct, not CBC. However the compatibility layer passes AES CBC flags. I pushed a better fix. Thanks

**Review by SparkiDev — APPROVED** [review]

**Pull request merged**